public/app/plugins/datasource/prometheus/querybuilder/components/LabelFilters.tsx

```diff
@@ -0,0 +1,81 @@
+import { isEqual } from 'lodash';
+import React, { useEffect, useState } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { EditorFieldGroup, EditorField, EditorList } from '@grafana/experimental';
+
+import { QueryBuilderLabelFilter } from '../shared/types';
+
+import { LabelFilterItem } from './LabelFilterItem';
+
+export const MISSING_LABEL_FILTER_ERROR_MESSAGE = 'Select at least 1 label filter (label and value)';
+
+export interface Props {
+  labelsFilters: QueryBuilderLabelFilter[];
+  onChange: (labelFilters: Array<Partial<QueryBuilderLabelFilter>>) => void;
+  onGetLabelNames: (forLabel: QueryBuilderLabelFilter) => Promise<SelectableValue[]>;
+  onGetLabelValues: (forLabel: QueryBuilderLabelFilter) => Promise<SelectableValue[]>;
+  /** If set to true, component will show error message until at least 1 filter is selected */
+  labelFilterRequired?: boolean;
+  getLabelValuesAutofillSuggestions: (query: string, labelName?: string) => Promise<SelectableValue[]>;
+}
+
+export function LabelFilters({
+  labelsFilters,
+  onChange,
+  onGetLabelNames,
+  onGetLabelValues,
+  labelFilterRequired,
+  getLabelValuesAutofillSuggestions,
+}: Props) {
+  const defaultOp = '=';
+  const [items, setItems] = useState<Array<Partial<QueryBuilderLabelFilter>>>([{ op: defaultOp }]);
+
+  useEffect(() => {
+    if (labelsFilters.length > 0) {
+      setItems(labelsFilters);
+    } else {
+      setItems([{ op: defaultOp }]);
+    }
+  }, [labelsFilters]);
+
+  const onLabelsChange = (newItems: Array<Partial<QueryBuilderLabelFilter>>) => {
+    setItems(newItems);
+
+    // Extract full label filters with both label & value
+    const newLabels = newItems.filter((x) => x.label != null && x.value != null);
+    if (!isEqual(newLabels, labelsFilters)) {
+      onChange(newLabels);
+    }
+  };
+
+  const hasLabelFilter = items.some((item) => item.label && item.value);
+
+  return (
+    <EditorFieldGroup>
+      <EditorField
+        label="Label filters"
+        error={MISSING_LABEL_FILTER_ERROR_MESSAGE}
+        invalid={labelFilterRequired && !hasLabelFilter}
+      >
+        <EditorList
+          items={items}
+          onChange={onLabelsChange}
+          renderItem={(item, onChangeItem, onDelete) => (
+            <LabelFilterItem
+              item={item as QueryBuilderLabelFilter}
```

Comment:
As a new item is created, all its properties will initially be empty.

This is a source of bugs here - if you call `item.label.indexOf()` in `<LabelFilterItem />` you'll crash because it didn't expect `item.label` to be undefined.

Reply:
`<LabelFilterItem />` should accept item will all it's properties being nullable (`Partial<>`), or, set defaults in here before passing it in.

Reply:
Yeah I thought that I was doing it a favor by removing some of those type assertions in the LabelFilterItem, but I agree that sets a bad expectation that all of the values have already been set. I went back to the pattern from the [original component](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/querybuilder/shared/LabelFilterItem.tsx). 

I spent a bit of time the other day trying to clean up all of the types in these "new" components, but it was looking like it would require some refactor of files not in scope for this PR. But I plan on spending some more time with it in a [follow up PR](https://github.com/grafana/grafana/issues/58375).